### PR TITLE
Add an option to compile with BinaryBuilder

### PR DESCRIPTION
**Pull request by amontoison**
An option to rule them all.

### CODE REVIEW

**Comment by amontoison** [comment]
@dpo 
It will help us to compile the artifacts `CUTEst_jll` and the future `GALAHAD_jll`.

File: compiler.binarybuilder.bb.fc [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+CUDACOMP=$FORTRAN
+CUDAARCH=
+CUDACODE=
+CUDABASIC=
+CUDALIBS=
+CUDAOPENMP=$OPENMP
+MOMP='-lgomp'
+TIMER=GEN
+BLAS='-lopenblas'
+LAPACK='-lopenblas'
```
Comment (dpo) [inline compiler.binarybuilder.bb.fc]: could these be LBT?

File: compiler.binarybuilder.bb.fc [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+HSL=
+SPRAL='-lgalahad_spral'
+METIS='-lgalahad_metis'
+PARDISO='-lgalahad_pardiso'
+MKL_PARDISO='-lgalahad_mkl_pardiso'
+WSMP='-lgalahad_wsmp'
+NOT95=IS95
+NOT64=IS64
+BINSHELL=sh
+SHARED='-shared'
```
Comment (dpo) [inline compiler.binarybuilder.bb.fc:46]: `-dynamiclib` on macOS.

**Review by dpo — COMMENTED** [review]
Interesting. Thanks!

File: compiler.binarybuilder.bb.fc (hunk 1 above)
Comment (amontoison) [reply compiler.binarybuilder.bb.fc]: Yes, LBT is just not working on Windows for Julia < v"1.9".
But GALAHAD is probably not working on Windows.

**Pull request merged**